src/com/example/Demo_Subj/RoomView.java

```diff
@@ -41,34 +162,207 @@ protected void onDraw(Canvas canvas){
+            for (Iterator<Room> iter = World.getAllRooms().iterator(); iter.hasNext(); ) {
+                Room room = iter.next();
+                if (room.getID() == aktRoomID) {
+                    drawRoom = room;
+                    drawRoomB = Bitmap.createScaledBitmap(
+                            BitmapFactory.decodeResource(resources, room.getPicResource()),
+                            GlobalInformation.getScreenWidth(),
+                            GlobalInformation.getScreenHeight(),
+                            false);
+                    lastRoom = room.getID();
+                }
             }
         }
+
         if (drawRoom == null) {
             // room not found
             drawRoom = roomList.get(0);
         }
 
+        //Animation für das Subjekt, übernommen aus Subjekt
+        /*****************************************************************************************/
+        intel.tick();                   //KI aufrufen
+        //hier muss später noch ein Aufruf zum ändern der Animation eingefügt werden, z.B. durch die KI oder durch RoomActivity selbst
+        switch (animation) {
+            case 0:     calculatePosition();
+                        break;
+
+            //Weiter Switch-Cases für weitere Animationen
+
+            default:    aktBitmap = subjStandBitmap;        //entweder das oder eine Fehlermeldung
+                        break;
+        }
+        //calculatePosition();
+
+        /*****************************************************************************************/
+
+
         // draw the background image
-        canvas.drawBitmap(drawRoom.getBitmapRoom(), 0, 0, p);
+        canvas.drawBitmap(drawRoomB, 0, 0, p);
 
         // draw all items (TODO: add layering)
-        List<Item> drawItems = drawRoom.getItemList();
-        
-        for (Iterator<Item> iter = drawItems.iterator(); iter.hasNext(); ){
-            Item obj = iter.next();
-            canvas.drawBitmap(obj.getBitmapO(), obj.getxPos(), obj.getyPos(), p);
+
+        for (Iterator<Item> iter = World.getAllItems().iterator(); iter.hasNext(); ){
+            Item item = iter.next();
+            int ID = item.getPicresource();
+            for (Iterator<Integer> id = drawRoom.getContainingitems().iterator(); id.hasNext(); ){
+                if(id.next() == ID){
+                    canvas.drawBitmap(
+                            BitmapFactory.decodeResource(resources, item.getPicresource()),
+                            item.getXPos().intValue(),
+                            item.getYPos().intValue(),
+                            p);
+                }
+            }
         }
 
-        // draw the subject
-        canvas.drawBitmap(subject.getSubjBitmap(),
-                subject.getxPos(),
-                subject.getyPos(), p);
+
+        canvas.drawBitmap(aktBitmap,
+                xPos,
+                yPos, p);
     }
 
+    private void calculatePosition(){
```

Comment:
Ich bin mir nicht sicher, wie das gehandlet werden sollte - eigentlich gehört das Ändern der Position mit ins Subjekt, nur die Darstellung (d.h. Grafik an die entsprechende Stelle setzen und animieren) hier in den Renderer. Das kann aber (falls es so erstmal geht) später gemacht werden. Die xPosition usw hat hier auf jeden Fall eigentlich nichts zu suchen.